kusd/validator/validator.go

```diff
@@ -31,8 +32,30 @@ type Backend interface {
 	ChainDb() kusddb.Database
 }
 
-// Validator represents a consensus validator
-type Validator struct {
+type Validator interface {
+	Start(coinbase common.Address, deposit uint64)
+	Stop() error
+	SetExtra(extra []byte) error
+	Validating() bool
+	SetCoinbase(addr common.Address) error
+	SetDeposit(deposit uint64)
+	Pending() (*types.Block, *state.StateDB)
+	PendingBlock() *types.Block
+	AddProposal(proposal *types.Proposal) error
+	AddVote(vote *types.Vote) error
+	AddBlockFragment(blockNumber *big.Int, round uint64, fragment *types.BlockFragment) error
+}
+
+var (
+	ErrCantStopNonStartedValidator       = errors.New("can't stop consensus, not started")
```

Comment:
just for consistency, shouldn't this be `not validating`?

Reply:
:+1:  true one or the other, not sure which though